src/PriorityQueue.js

```diff
@@ -1,52 +1,76 @@
-'use strict'
-
-//implement using heap
-
-const _ = require('lodash')
-const Heap = require('./Heap.js')
-const Item = require('./Item.js')
-
-class PriorityQueue {
-  constructor(comparator) {
-    if (!comparator)
-      throw new Error('need to provide comparator')
-    this.heap = new Heap(comparator)
-  }
```

Comment:
How does removing the class help? Isn't it better to keep the class?

Reply:
I generally prefer pure data oriented design. Better performance, less memory used. Also removes the need for sprinkling `this.` keyword everywhere.

Reply:
I am concerned that it changes the existing api and breaks the existing code. Can we create a different file and reuse components to expose both ways?

Reply:
We are breaking the API anyway by introducing a `maxLength` argument in the PriorityQueue's constructor function. If you decide to merge this PR it should be released to npm as 2.0.0.

Reply:
We aren't breaking the api as it is an optional parameter so existing api should work as is. If we look at it in an object oriented way it is better to keep the class.

Reply:
If it is object oriented or functional, either way the AP breaks because there is a finite length on the heap size now. If you don't specify the `maxLength` parameter it defaults to 1000. That means any existing users that have more than 1000 entries in their heap will see their code stop working.

Reply:
Let's make it 5000 then? Their tests should break if they upgrade the api.

Reply:
If you make it 5000, what happens if an existing user tries to queue 5001 items?

There is no way to avoid the fact that this change will require updating the version to `2.0.0` because it's going to impact some users.

There are also other changes in this PR that are breaking (allowing -1 as a return value, adding an optional `dequeueHighest` argument.)

Reply:
The assumption is that if we make it 5001 there will be less users who face the issue then. Yes it is not a way around updating to the next major version.